dts/bindings/pwm/nxp,flexio-pwm.yaml

```diff
@@ -0,0 +1,64 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: NXP Flexio PWM controller.
+  Each flexio timer can be used for generating one pwm pulse.
+  The two PWM modes supported by flexio are chosen based on the selected polarity -
+  Dual 8-bit counters PWM mode and Dual 8-bit counters PWM Low mode.
+
+compatible: "nxp,flexio-pwm"
+
+include: [pwm-controller.yaml, pinctrl-device.yaml, base.yaml]
+
+properties:
+  pinctrl-0:
+    required: true
+
+  pinctrl-names:
+    required: true
+
+  "#pwm-cells":
+    const: 3
+
+pwm-cells:
+  - channel
+  # Period in terms of nanoseconds
+  - period
+  - flags
+
+child-binding:
+  description: |
+    Flexio PWM channel configuration.
+
+  properties:
+    pin-id:
+      type: int
+      required: true
+      description: |
+        pin-id should be populated with number 'n' specified in the
+        FXIO_D<n> data pin of your board's schematic.
+
+    polarity:
```

Comment:
We could instead define this as a string enum, like so:
```
    polarity:
      type: string
      enum:
        - "active-low"
        - "active-high"
      description: |
        Output polarity for PWM channel.
```

`DT_ENUM_IDX` could then be used to convert the polarity selection to an integer value

Reply:
Done !!